pkg/sql/schemachanger/scplan/internal/rules/dep_add_index.go

```diff
@@ -71,21 +71,36 @@ func init() {
 func init() {
 
 	registerDepRule(
-		"index named right before index becomes public",
+		"primary index named right before index becomes public",
 		scgraph.SameStagePrecedence,
 		"index-name", "index",
 		func(from, to nodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.Type((*scpb.IndexName)(nil)),
 				to.Type(
 					(*scpb.PrimaryIndex)(nil),
-					(*scpb.SecondaryIndex)(nil),
 				),
 				joinOnIndexID(from, to, "table-id", "index-id"),
 				statusesToPublicOrTransient(from, scpb.Status_PUBLIC, to, scpb.Status_PUBLIC),
 			}
 		},
 	)
+
+	registerDepRule(
+		"secondary index named before validation",
+		scgraph.Precedence,
+		"index-name", "index",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.Type((*scpb.IndexName)(nil)),
+				to.Type(
+					(*scpb.SecondaryIndex)(nil),
+				),
+				joinOnIndexID(from, to, "table-id", "index-id"),
+				statusesToPublicOrTransient(from, scpb.Status_PUBLIC, to, scpb.Status_WRITE_ONLY),
```

Comment:
Why is the `to` status not `scpb.Status_VALIDATED` here? Probably nothing bad happens from it being `scpb.Status_WRITE_ONLY` but it's a bit weird.